Remove leading/trailing whitespaces from rendered authorization header

kyri-petrou opened the pull request:
/fixes #2742
/claim #2742

Seems that Netty doesn't like the leading whitespace in the Authorization header, which is causing the client to hang

kyri-petrou committed:
Fix client hanging when Authorization header contains trailing whitespace

gaeljw commented:
Interesting, could it be an issue in Netty then? Maybe worth letting them know?

987Nabil commented:
@kyri-petrou is line 1064 not a problem too? We split by space and then scheme is index 0 of that array. But if there is no scheme, index 0 would be params not scheme. For the stringification, I think it does not matter. But the `Unparsed` instance would be wrong or? I think if the splitted result has size one, scheme should be empty.

kyri-petrou committed:
Allow parsing Authorization header without scheme

kyri-petrou committed:
Filter empty strings when parsing the authorization header

kyri-petrou commented:
> @kyri-petrou is line 1064 not a problem too? We split by space and then scheme is index 0 of that array. But if there is no scheme, index 0 would be params not scheme. For the stringification, I think it does not matter. But the `Unparsed` instance would be wrong or? I think if the splitted result has size one, scheme should be empty.

Good points. I did the following:
1. Removed any empty strings from the array after splitting on `" "`
2. Parsing a single non-empty string to `Unparsed("", value)`
3. Otherwise parse the first value as the scheme and then the rest as previously
